react/src/hooks/useLoadAndSetGlobalPreferences.ts

```diff
@@ -0,0 +1,94 @@
+import { useContext, useEffect, useState } from "react";
+import { useAuth0 } from "@auth0/auth0-react";
+import { gql, useLazyQuery } from "@apollo/client";
+import { useLocation, useNavigate } from "react-router-dom";
+import { GlobalPreferencesContext } from "providers/GlobalPreferencesProvider";
+import { ApolloAuth0WrapperContext } from "providers/ApolloAuth0Provider";
+import { BasesQuery } from "types/generated/graphql";
+
+export const useLoadAndSetGlobalPreferences = () => {
+  const { user } = useAuth0();
+  const navigate = useNavigate();
+  const location = useLocation();
+  const { globalPreferences, dispatch } = useContext(GlobalPreferencesContext);
+  const { isAccessTokenInHeader } = useContext(ApolloAuth0WrapperContext);
```

Comment:
Is this the key to the 401s due to missing tokens in the header that we saw?

Reply:
okay yep apparently: https://github.com/boxwise/boxtribute/pull/704/files#diff-dff76fc64a9d7628c33c39e139bc4fd76c3f60a138b28ba14d1a23d4f0c98802R101